### ISSUE TITLE
ECS logging format

### ISSUE THREAD

**Issue by jehne** [keycloak/keycloak#33959]
### Description

It would be useful to have keycloak capable of logging into ECS format, when logging to the console. (https://www.elastic.co/guide/en/ecs/current/ecs-log.html)

### Motivation

ESC logging format is useful eg. if you want to integrate Keycloak with Elastic observability feature.

### Details

As a PoC I have just added the dependency: 
```
 <dependency>
            <groupId>io.quarkiverse.loggingjson</groupId>
            <artifactId>quarkus-logging-json-deployment</artifactId>
            <version>3.1.0</version>
        </dependency>
```
into following maven modules  `keycloak-quarkus-server`, `keycloak-quarkus-server-deployment`

and then add `quarkus.log.json.log-format=ecs` into quarkus.properties.

The result looks like:
```
my@laptop:~/BRF/keycloak-26.0.0$ ./bin/kc.sh start-dev
Updating the configuration and installing your custom providers, if any. Please wait.
2024-10-15 16:08:04,432 INFO  [io.qua.dep.QuarkusAugmentor] (main) Quarkus augmentation completed in 6671ms
{"@timestamp":"2024-10-15T16:08:08.013+02:00","log.logger":"org.keycloak.quarkus.runtime.storage.infinispan.CacheManagerFactory","log.level":"INFO","process.thread.name":"main","process.thread.id":1,"host.name":"petr-thinkpad-e15-gen-3","message":"Starting Infinispan embedded cache manager","ecs.version":"1.12.1"}
{"@timestamp":"2024-10-15T16:08:08.193+02:00","log.logger":"org.keycloak.quarkus.runtime.storage.infinispan.CacheManagerFactory","log.level":"INFO","process.thread.name":"main","process.thread.id":1,"host.name":"petr-thinkpad-e15-gen-3","message":"Persistent user sessions enabled and no memory limit found in configuration. Setting max entries for sessions to 10000 entries.","ecs.version":"1.12.1"}
.
.
.
```
This is just a PoC and should be available through standard configuration way, eg. over the `log-console-output=ecs` or another config property.

Thanks a lot.

**Comment by mabartos** [comment]
@jehne Thanks for creating this issue. However, we're not able to include any Quarkiverse dependency on Keycloak by default due to productization issues and lack of support from Quarkus. 

As you mentioned, you can build your own Keycloak with all the necessary changes. 

JFYI - I'm working on a tool, that can be used for including any Quarkus/Quarkiverse extension to your Keycloak deployment that can even be part of your Keycloak docker image: https://github.com/mabartos/keycloak-quarkus-extensions. Let me know if you'd be interested in it.

**Comment by jehne** [comment]
@mabartos thank you for the information. Sure I am interested about your tool :) but I can't find any "how to" in repo. Could you please provide any, or describe, how to use https://github.com/mabartos/keycloak-quarkus-extensions ? Thank you.

**Comment by mabartos** [comment]
@jehne I started yesterday, so no big progress yet - but it's already possible to do it. I'll try to add all the necessary information by the end of this or next week.

**Comment by mabartos** [comment]
@bernhof Ahh, sorry, didn't post the progress around that. It is possible to set ECS via this tool[1] for now. As mentioned in the #36854, we should be ready to provide support for ECS by default - but it needs to be verified.

[1] https://github.com/mabartos/keycloak-quarkus-extensions/blob/main/examples/logging-json-ecs.md

**Comment by mabartos** [comment]
ECS support will be available in Keycloak 26.2.0 - https://github.com/keycloak/keycloak/issues/36854
